### ISSUE TITLE
WebClient like API using HttpClient

### ISSUE THREAD

**Issue by ChristianKern97** [dotnet/core#5039]
# WebClient like API for HttpClient

As `WebClient` is an obsolete API (as far as I got told) and has some problems ([i.e. automatic redirecting](https://stackoverflow.com/questions/63327319/webclient-is-not-following-redirects-in-net-core-3-1)) it would be nice to have a similar replacement API using the HttpClient

# Problem

So, a few people have told me, that the `WebClient` API is marked as obsolete and the `HttpClient` should be used instead. The problem is that creating an asynchronous download with progress reports, and other very helpful stuff, is very easy with `WebClient`, but some messy code with `HttpClient`.

For this reason, most people will use the `WebClient` API and encounter the same problems over and over again.

# Solution

There are two possible solutions: 
- Fix the `WebClient` API
- Implement a similar API with similar functionalities but based on the `HttpClient`

**Comment by GeraudFabien** [comment]
Webclient would be fix just for Progress report wich as already solution like ProgressMessageHandler and where you may thinks about using websocket or split into chunck.

**Comment by carlossanlop** [comment]
@scalablecory can you follow up?

**Comment by Neme12** [comment]
I agree that progress report would be useful. It's currently really hard, and you have to implement everything yourself, including a custom stream copy with progress.

**Comment by Neme12** [comment]
This issue should probably be moved to https://github.com/dotnet/runtime.
